src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,32 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_AMOUNT_OF_AGE = 18;
+    private static final int MINIMAL_AMOUNT_OF_SYMBOLS = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("You can not add user without any parameters");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with such login already exists");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < MINIMAL_AMOUNT_OF_SYMBOLS) {
+            throw new RegistrationException("Login should have at least 6 characters");
```

Comment:
better pass constant to message 
```suggestion
            throw new RegistrationException("Login should have at least " + MINIMAL_AMOUNT_OF_SYMBOLS + '' characters");
```
when you choose to change this requirement, you will only need to change the value of the constant, the messages will be valid (fix other messages)